src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -179,6 +191,18 @@ protected Object invokeStep(StepDescriptor d, Object args) {
             d.checkContextAvailability(context);
             Thread.currentThread().setContextClassLoader(CpsVmExecutorService.ORIGINAL_CONTEXT_CLASS_LOADER.get());
             s = d.newInstance(ps.namedArgs);
+            if (isKeepStepInfo() && !(s instanceof ParallelStep)) { // Storing ArgumentsActions or not
+                Computer comp = context.get(Computer.class);
+                if (comp != null) {
+                    // Get the environment variables to find ones that might be credentials bindings
+                    // But filter out variables coming from the host itself
+                    EnvVars allEnv = new EnvVars(context.get(EnvVars.class));
+                    allEnv.entrySet().removeAll(comp.getEnvironment().entrySet());
+                    an.addAction(new ArgumentsActionImpl(ps.namedArgs, allEnv));
+                } else {
+                    an.addAction(new ArgumentsActionImpl(ps.namedArgs));  // No EnvVars that can supply credentials bindings
```

Comment:
:bug: Not true after https://github.com/jenkinsci/credentials-binding-plugin/pull/34 landed